icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -0,0 +1,457 @@
+
+
+    TTree *fEventTree;
+    std::vector<TTree*> fOpFlashTrees;
+    std::vector<TTree*> fOpHitTrees;
+
+    int m_run;
+    int m_event;
+    int m_timestamp;
+    //int m_nflashes;
+    //int m_nophit;
+    float m_beam_gate_start=-99999;
+    float m_beam_gate_width=-99999;
+    int m_beam_type=-1;
+    int m_flash_id;
+    int m_multiplicity;
+    int m_multiplicity_left;
+    int m_multiplicity_right;
+    float m_sum_pe;
+    float m_sum_pe_left;
+    float m_sum_pe_right;
+    float m_flash_time;
+    //float m_flash_x;
+    //float m_flash_width_x;
+    float m_flash_y;
+    float m_flash_width_y;
+    float m_flash_z;
+    float m_flash_width_z;
+
+    int m_channel_id;
+    float m_integral; // in ADC x tick
+    float m_amplitude; // in ADC
+    float m_start_time;
+    float m_width;
+    float m_abs_start_time;
+    float m_pe;
+    float m_fast_to_total;
+
+    std::vector<float> m_pmt_x;
+    std::vector<float> m_pmt_y;
+    std::vector<float> m_pmt_z;
+
+};
+
+
+opana::ICARUSFlashAssAna::ICARUSFlashAssAna(Parameters const& config)
+  : EDAnalyzer(config)
+  , fTriggerLabel( config().TriggerLabel() )
+  , fOpHitLabel( config().OpHitLabels() )
+  , fFlashLabels( config().FlashLabels() )
+  , fPEOpHitThreshold( config().PEOpHitThreshold() )
+{ }
+
+
+void opana::ICARUSFlashAssAna::beginJob() {
+
+  art::ServiceHandle<art::TFileService const> tfs;
+
+  TTree* fGeoTree = tfs->make<TTree>("geotree", "geometry information" );
+  fGeoTree->Branch("pmt_x",&m_pmt_x);
+  fGeoTree->Branch("pmt_y",&m_pmt_y);
+  fGeoTree->Branch("pmt_z",&m_pmt_z);
+  
+  auto const geop = lar::providerFrom<geo::Geometry>();
+
+  double PMTxyz[3];
+  for(size_t opch=0; opch<geop->NOpChannels(); ++opch) {
+
+    geop->OpDetGeoFromOpChannel(opch).GetCenter(PMTxyz);
+
+    //std::cout << PMTxyz[0] << " " << PMTxyz[1] << " " << PMTxyz[2] << std::endl;
+
+    m_pmt_x.push_back(PMTxyz[0]);
+    m_pmt_y.push_back(PMTxyz[1]);
+    m_pmt_z.push_back(PMTxyz[2]);
+
+  }
+
+  fGeoTree->Fill();
+
+  fEventTree = tfs->make<TTree>("eventstree", "higher level information on the event" );
+  fEventTree->Branch("run", &m_run, "run/I");
+  fEventTree->Branch("event", &m_event, "event/I");
+  fEventTree->Branch("timestamp", &m_timestamp, "timestamp/I");
+  //fEventTree->Branch("nflashes", &m_nflashes, "nflashes/I");
+  //fEventTree->Branch("nophits", &m_nophit, "nophits/I");
+  fEventTree->Branch("beam_gate_start", &m_beam_gate_start, "beam_gate_start/F");
+  fEventTree->Branch("beam_gate_width", &m_beam_gate_width, "beam_gate_width/F");
+  fEventTree->Branch("beam_type", &m_beam_type, "beam_type/I");
+
+  if ( !fFlashLabels.empty() ) {
+
+    for( auto const & label : fFlashLabels ) {
+
+        // TTree for the flash in a given cryostat
+        std::string name = label.label()+"_flashtree";
+        std::string info = "Three for the recob::Flashes with label "+label.label();
+
+        TTree* ttree = tfs->make<TTree>(name.c_str(), info.c_str() );
```

Comment:
So when I make a TTree like this I always make the branch definition as a C++ class with a ROOT dictionary. Is there a preference in style between doing that / making custom TBranch's as is done here?